zarr/tests/test_core.py

```diff
@@ -949,6 +949,15 @@ def test_dtypes(self):
             z[:] = a
             assert_array_almost_equal(a, z[:])
 
+        # complex
+        for dtype in 'c8', 'c16':
+            z = self.create_array(shape=10, chunks=3, dtype=dtype)
+            assert z.dtype == np.dtype(dtype)
+            a = np.linspace(0, 1, z.shape[0], dtype=dtype)
+            a -= 1j * a
+            z[:] = a
+            assert_array_almost_equal(a, z[:])
```

Comment:
Given the special treatment required for fill values, would it make sense to add an explicit check that complex data with missing values is handled correctly? I see that the metadata encoding is checked below, but not the actual filling procedure (as far as I can tell).

Reply:
Could you please elaborate a bit on what kind of representation of missing data we are discussing? We handle `NaN` currently, but maybe this is not what you are thinking about.

Reply:
What I mean is to make sure that missing data (the kind that triggers the `fill_value` machinery to be activated) is present in the array. For example, by adding
```python
a[0] = np.nan
```

However, I might be understanding what `fill_value` does in zarr. Is `fill_value` only used to represent uninitialized portions of the array? Or are all nans explicitly translated to `fill_value` before serialization?

Reply:
AFAIK `fill_value` is only used to handle uninitialized chunks. Am not aware of any special behavior around `NaN`s in Zarr. Does that help?